index.js

```diff
@@ -0,0 +1,74 @@
+var _ = require('underscore');
+
+var SolidusAssetsProxy = function(proxied_assets) {
+  if (!(this instanceof SolidusAssetsProxy)) return new SolidusAssetsProxy(proxied_assets);
+
+  this.proxied_assets = proxied_assets.map(function(assets) {
+    // Match URL strings from origin up to the query string or ending delimiter
+    var regexp = _.isRegExp(assets.origin) ? assets.origin.source : escapeRegExp(assets.origin);
+    return {
+      origin: new RegExp('(^|[\'"\\s\\(])(' + regexp + '.*?)($|[\\?\'"\\s\\)])', 'g'),
+      proxy: addSlash(assets.proxy)
+    };
+  });
+};
+
+SolidusAssetsProxy.prototype.proxyAssets = function(string) {
+  if (!string || !string.substring) return string;
+
+  return this.proxied_assets.reduce(function(result, assets) {
+    return result.replace(assets.origin, function(match, starting_delimiter, asset, ending_delimiter) {
+      return starting_delimiter + assets.proxy + encodeURIComponent(asset) + ending_delimiter;
+    });
+  }, string);
+};
+
+SolidusAssetsProxy.prototype.proxyContextAssets = function(context, paths) {
+  if (paths) {
+    for (var i in paths) {
+      proxyObjectAssetsByPath.call(this, context, paths[i].split('.'));
+    }
+  } else {
+    proxyObjectAssets.call(this, context, true);
+  }
+};
+
+module.exports = SolidusAssetsProxy;
+
+// PRIVATE
+
+function escapeRegExp(string) {
+  return string.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
+};
+
+function addSlash(string) {
+  return string.slice(-1) == '/' ? string : string + '/';
+};
+
+function proxyObjectAssetsByPath(object, path) {
+  if (!object) return;
+  if (_.isArray(object)) {
```

Comment:
Any reason you're not using `Array.isArray` beyond uniformity with the other checks?